linux/compat.bsh

```diff
@@ -261,40 +289,58 @@ function load_vsi_compat()
   #    # x will be 12 when bash_bug_local_shadow_exported_variable is 1
   #    export -f foo; x=11 bash -c foo # Same result as above
   #    x=13 foo # x is always 13 and in the bash child too. No explanation for this behavior
+  #
+  # .. var:: bash_bug_allocate_xtracefd
+  #
+  # In Bash 4.1 and 4.2, there is a bug where the following command does not work right:
+  #
+  # .. code-block:: bash
+  #
+  #    exec {BASH_XTRACEFD}>"somefile"
+  #
+  # ``BASH_XTRACEFD``'s value is set, but whatever triggers ``BASH_XTRACEFD`` to work, is not triggered.
+  #
+  # .. note::
+  #
+  #   ``BASH_XTRACEFD`` and automatically allocation file descriptors are not available features in bash 4.0 or older, but :var:`bash_bug_allocate_xtracefd` will still be set to ``0``
```

Comment:
automatically allocated file (?)

Reply:
It's another bash feature, dictated by `bash_feature_allocate_file_descriptor`

```
 Each  redirection  that  may  be  preceded by a file descriptor number may instead be preceded by a word of the form {varname}.  In this case, for each redirection operator except >&- and <&-, the shell will allocate a file descriptor greater than or equal to 10 and assign it to varname.  If >&- or <&- is preceded by {varname}, the value of varname defines the file descriptor to close.   If {varname} is supplied, the redirection persists beyond the scope of the command, allowing the shell programmer to manage the file descriptor himself.
```